builtin/docker/registry_docker.go

```diff
@@ -83,7 +122,7 @@ func (r *Registry) pushWithDocker(
 		// Resolve the Repository name from fqn to RepositoryInfo
```

Comment:
It looks like this is the only part of the `findAuth` function that's currently being used in `pullImage`, because you're not setting `r.config.EncodedAuth` or `r.config.Password`.